spec.md

```diff
@@ -200,6 +200,16 @@ that contains both context and data).
   * If present, MUST adhere to the format specified in
     [RFC 3986](https://tools.ietf.org/html/rfc3986)
 
+### sampled-rate
+* Type: Positive Integer
+* Description: The rate at which this event has already been sampled. Represents
+  the denominator of a fraction: when `1/n` events are sent, this field holds
```

Comment:
would we need to say what the time-frame of this sample is? seconds? minutes?

Reply:
I haven't needed the timeframe in the past. The way I've used this field in Scuba (the technology HoneyComb is based on), the sample effectively means "weigh this data point N times heavier". The meaning of that weight depends on whether the summary is summing, averaging, etc.

Reply:
Sorry for being dense, but "N times heavier" than what? Got a link to Scuba so I can see how they use it?

Reply:
In the most general sense, a sample rate of 5 means that this event represents 5 events - that the system originating the events chose not to send 4 of them and is sending this one saying "there were actually 5, but I'm just giving you this one." It doesn't say anything about the time range across which those other events may have occurred. When taking an action based on this event, the target system then has enough information to extrapolate based on the received event and take an action appropriate for 5 events, despite having received only one.

Presenting a graph based on the content of the event is an easy way to describe what this looks like in practice - if one were to ask "how many events happened", the total count would have to be scaled according to the sample rate of each event received.  How this calculation plays out depends on the action to take (a count of events is obvious, an average or percentile is a little more complicated, but not terribly so) as well as the goals of the receiving system. The description "weight this data point N times heavier" is clear in the case of calculating an average - the sampled data point should effectively count N times towards the average instead of just once.

The Scuba whitepaper does mention sampling, but doesn't cover it in great detail.  Here's a link to the paper: https://research.fb.com/publications/scuba-diving-into-data-at-facebook/

Reply:
@maplebed thanks - that helps a ton.  My head was stuck in a different definition.

@inlined what do you think about this tweak to the text:
- - -
The rate at which this event had been sampled. This value represents
the number of times this event occurred.
- - -

Reply:
@maplebed any comment?

Reply:
The prose sounds fairly good. I think we can improve the description a bit further if we describe what scope is being sampled. E.g. if I ask for sampled events from MySQL, is the sampling based on row, table, or database? That would certainly impact any sort of visualization or metrics I wanted to generate.

Reply:
@duglin Yeah, that wording change LGTM. I'll update the diff. 
@inlined what sort of events are you imagining you're getting from a MySQL database? That seems to be a larger influence on how sampling is done.  The events I usually expect to get from my DB are "a query happened" (whether that was a read or write request) and the sampling would clearly be "I'm sending you 1 out of the 5 queries that happened", and not anything about rows or tables or databases (beyond that a specific query will be on a database/table).  You can always choose a sample rate based on something more specific, of course, and count up reads / writes differently, choosing to emit different numbers of each (all writes and 1/10 reads?) by setting the `sampled-rate` field appropriately on each event you emit.

Reply:
hmm...  thinking about this a bit more, "This value represents the number of times this event occurred." sounds more like a counter that one might imagine only increments.  Maybe "This value represents the number of similar events that were not sent"?

Reply:
Changed the wording with a8cf97d63 - let me know what you think.